### ISSUE TITLE
yellow or red alert for 'Legacy'

### ISSUE THREAD

**Issue by grahamperrin** [mozilla/addons-frontend#3780]
For users of 56.x – and [until mid 2018](https://wiki.mozilla.org/RapidRelease/Calendar#Future_branch_dates), users of ESR 52.x: 

- there should be an upfront indication wherever an extension is legacy. 

## Precedents

Established use of a yellow rectangle for LEGACY at `about:addons` within the browser. 

The subtle new look yellow alert alongside _Restart required_ at, for example, https://addons.mozilla.org/addon/quick-locale-switcher-2/

## Thoughts

The requirement for _Legacy_ extensions to be **clearly identifiable** as such probably arises primarily from the new look disappearance of clear identification (green badge) wherever an extension will be compatible with _Firefox 57+_

As there's space for the subtle yellow ⟲ for the relatively rare extensions that will require a restart of Firefox, there must be space for an unmistakable yellow alert for the relatively commonplace extensions that will be unusable with Firefox 57 and greater (all of which will, I guess, disappear from AMO sometime in Q4 2018).

It's highly desirable to have the distinction at a time of **increased likelihood of people 'shopping'** for extensions ahead of the mid-November release of 57. 

### Urgency

If you can (please): 

- Monday 2017-11-06 for a Tuesday freeze and push on Thursday 2017-11-09

– TIA.  

## Related

https://www.reddit.com/r/firefox/comments/7adng2/did_the_addon_page_change/dp9833c/ (9 points, recently up from 7):

> How can I see if an addon is FF 57+ ready now?

https://www.reddit.com/r/firefox/comments/7adng2/did_the_addon_page_change/dp95kcl/ (17 points, rising): 

> … can't tell anymore … 57 or not.

**Comment by Mark12547** [comment]
###  Experience without any WebExtensions / LEGACY indicator:

I lived through missing WebExtensions indication on AMO during late Nightly 56 / early Nightly 57, finding 57+ compatible extensions by searching AMO frustrating at that time, having to install the extension before finding out whether or not it was LEGACY; but the frustration level eased up tremendously when we got "COMPATIBLE WITH FIREFOX 57+" on AMO for extensions that were pure WebExtensions.

### User Foresight: Find WebExtensions Before 57 Lands, vs. Still LEGACY

Since preparation can often reduce the pain, some Firefox 56 users (and probably next summer some Firefox 52 ESR users) are trying to find WebExtensions replacements for their LEGACY extensions; having 57-compatible extensions already installed and having replaced LEGACY extensions gives time to get used to the new extensions and, when 57 lands in Release, it should produce a much smoother transition, and some might not even notice at first that Release had been upgraded.

But if the user is unprepared and has a bunch of LEGACY extensions, the 57 upgrade will cause all those LEGACY extensions to suddenly stop working. For some, this will come at the worst time, when they need to get their job done and don't have time to search for replacements. Those users are likely to experience lots of frustration.

### My Concern

Therefore, I am _very concerned_ about the absence of any indicator (whether "COMPATIBLE WITH FIREFOX 57+" or "LEGACY") when a Firefox Release 56 or Firefox ESR 52 user searches for a WebExtensions extension to replace a LEGACY extension because, as it is now, the new AMO interface gives no indication to a Firefox 56 user whether an extension is LEGACY (and will stop working in 10 days) or if it is pure WebExtensions (and will continue working for the foreseeable future). Those users will experience a good part of the frustration I had experienced last summer when my Nightly was going through the 56 &rarr; 57 transition and, worse, seek out other browsers that don't give them that sudden frustration.

### A Work-around

At the moment, when I direct Firefox 56 users to AMO, I have to tell them to click on "View classic desktop site" so they can see which extensions will continue working when 57 lands. While it works, it is less than ideal. It's also a message that many Firefox users won't be aware of as they Google and find answers that worked before the latest AMO interface changes.

### Wrap-up

As I mentioned in the first paragraph, I found the lack of a WebExtensions or LEGACY indicator on AMO frustrating. We want to *learn* from that experience, not *force Firefox users to relive that experience*. I really hope Firefox 56 users and Firefox 52 ESR users will soon have an indication on AMO (we have *learned* from my experience), rather than working with missing information (those users having to *relive* my experience).

Thank you!

**Comment by Mark12547** [comment]
Also, in Firefox 56.0.2, about:addons &rarr; "Extensions", the legacy extensions installed have a yellow "LEGACY" warning. Clicking on that label takes them to https://support.mozilla.org/en-US/kb/firefox-add-technology-modernizing?as=u&utm_source=inproduct that includes this bolded text:

> **We recommend that you install add-ons with the "Compatible with Firefox 57+" label to ensure compatibility with Firefox beyond November 2017.**

Unfortunately, following the link to AMO from that page and then searching for an extension winds up with extension description pages that give no clue whether or not those extensions are "Compatible with Firefox 57+", which will lead to further user frustration, especially if they install a new extension to replace their old LEGACY extension and then find out 10 days later that extension will likewise cease to operate.

**Comment by grahamperrin** [comment]
Thanks @Mark12547: 

- Mozilla bug [1414626 - kb/firefox-add-technology-modernizing outdated direction to seek the "Compatible with Firefox 57+" label on addons.mozilla.org (AMO)](https://bugzilla.mozilla.org/show_bug.cgi?id=1414626)

I have marked the knowledge base article as confusing, with reference to bug 1414626. Also I'll ping someone on the SUMO side of things.

**Comment by muffinresearch** [comment]
I'm marking this as p1 and I'll raise the question with our product people to determine if we can mirror the legacy labelling in Firefox rather than follow the same labelling used on the old AMO frontend or vice versa.

**Comment by willdurand** [comment]
Say we want to show a "not compatible with FF 57+" badge, we would need an icon for that (we only have an icon for "compatible with FF 57+" but not the opposite).

Only web extension add-ons are compatible with FF 57+ or is there something else?

**Comment by grahamperrin** [comment]
> **`Not compatible with Firefox 57+`**

Thought-provoking, thanks. It's precise, but maybe visually too wide for some purposes. 

Reading this alongside chat in IRC, first a bunch of **56.0.2** screenshots then I'll add a link to a very brief screen recording. 

tl;dr toying with the idea of an icon, with some yellow, to imply that something will _stop_ working (I do already have in mind an icon, but (for myself) I need to post this, step away from the computer to the kitchen for a few minutes then come back to get some perspective) …

![2017-11-06 18 07 49 mozilla firefox start page - mozilla firefox](https://user-images.githubusercontent.com/192271/32458157-63b8f906-c323-11e7-96ac-cf5cca560a90.png)

![2017-11-06 18 09 58 add-ons manager - mozilla firefox](https://user-images.githubusercontent.com/192271/32458162-67fccba0-c323-11e7-8231-cb393ae41b4f.png)

![2017-11-06 18 11 24 add-ons manager - mozilla firefox](https://user-images.githubusercontent.com/192271/32458215-936c4ff4-c323-11e7-90f9-fc9193b9e61f.png)

![2017-11-06 18 11 40 add-ons manager - mozilla firefox](https://user-images.githubusercontent.com/192271/32458165-6b913896-c323-11e7-9243-6389181c3718.png)

![2017-11-06 18 30 09 rss-feeds nachrichten bloggen add-ons fur firefox](https://user-images.githubusercontent.com/192271/32458171-6f8facd4-c323-11e7-9e54-a60024879e68.png)

![2017-11-06 18 31 53 aktuell beliebte erweiterungen add-ons fur firefox](https://user-images.githubusercontent.com/192271/32458180-732d7c40-c323-11e7-88a9-220791b5511a.png)

![2017-11-06 18 39 suchergebnisse fur stop add-ons fur firefox](https://user-images.githubusercontent.com/192271/32458181-7685d7b6-c323-11e7-9c04-3f8e95bb7ade.png)

![2017-11-06 18 40 tag firefox57 suchergebnisse fur stop add-ons fur firefox](https://user-images.githubusercontent.com/192271/32458185-7a7667dc-c323-11e7-996a-7c5e0d77a3ca.png)

**Comment by muffinresearch** [comment]
For the record @willdurand here's a copy of the icon [1]  - the copy as provided by @devaneymoz is "Not compatible with Firefox Quantum"

[1] [not-compatible-min.svg.zip](https://github.com/mozilla/addons-frontend/files/1447419/not-compatible-min.svg.zip)

**Comment by grahamperrin** [comment]
Thanks! Exported lazily to PNG (not for reuse): 

![not-compatible-min svg-exported](https://user-images.githubusercontent.com/192271/32458927-e2955ba0-c325-11e7-9716-080c19165c9c.png)

Consider the possibility/likelihood that this will be confused with other meanings of the red no entry icon. In particular, the fifth screenshot above (the first in German) where I point at a **red no entry icon for a greyed-out extension that bears the green badge of compatibility with Firefox 57+**. Yep, I'm focusing on the visuals without paying attention to the words …

**Comment by grahamperrin** [comment]
Extensions for Firefox: a new-look scroll through two pages of 'Stop…' -named extensions
https://www.youtube.com/watch?v=lhvBq8yNW2w (fifteen seconds)

**Comment by grahamperrin** [comment]
`/me` finishes thinking aloud in IRC. Quote: 

> … gut feeling about the red no entry icon is that it'll be not as effective as … say, a yellow 'stop' hand (palm towards the viewer).
> 
> There might be a whole lotta red going on already when e.g. people hit some of the key pages. Like, AdBlock and its variants. Yes it's a no entry sign and its red but there's some subconscious tuning-out from the red, I reckon. Consider https://addons.mozilla.org/firefox/ …
> 
> … speeds through https://addons.mozilla.org/firefox/search/?&sort=relevance&type=persona&q=yellow https://addons.mozilla.org/firefox/search/?&sort=relevance&type=persona&q=summer https://addons.mozilla.org/firefox/search/?&sort=relevance&type=persona&q=red https://addons.mozilla.org/search/?sort=relevance&type=extension&q=YouTube and https://addons.mozilla.org/search/?sort=relevance&type=extension&q=blocker

**Comment by Mark12547** [comment]
If I were looking for a suitable icon for NOT COMPATIBLE WITH 57+, I think it would be the Not (circle with diagonal line) over 57+. It would also be nice to have a mouse hover hint that would have a bit longer explanation if the icon alone doesn't convey enough information to the particular individual.

**Comment by grahamperrin** [comment]
>> … as effective as … say, a yellow 'stop' hand (palm towards the viewer). … 

(I [changed my mind about yellow](https://github.com/mozilla/addons-frontend/pull/3807#issuecomment-342271723) when I saw contexts – another, more significant, adjacent use of yellow. Red is fine.)

**Comment by ValentinaPC** [comment]
Verified this on AMO-dev FF56(Win 7).
The new label was added.
As the issue #3814 was logged separately, I think we can mark this as verified-fixed.
Postfix screenshot:
![2017-11-07_1229](https://user-images.githubusercontent.com/15685960/32489427-9ddcc47e-c3b8-11e7-8ad3-a21815572e00.png)
![2017-11-07_1231](https://user-images.githubusercontent.com/15685960/32489428-9dfe3cb2-c3b8-11e7-8419-0587d85176b4.png)
